### ISSUE TITLE
copying manifold ids and extract_boundary_mesh

### ISSUE THREAD

**Issue by drwells** [dealii/dealii#16392]
Fall-out from #11566.

The following doesn't work any more:
1. create a mesh with a `Manifold`
2. refine that mesh
3. extract a boundary mesh

since `extract_boundary_mesh()` needs manifolds to do mesh refinement, but those aren't available at this point.

I'm not sure if we should just ban extracting refined boundary meshes or if we should use some other approach to support this use case.

**Comment by peterrum** [comment]
I am wondering whether it makes sense to extract a boundary mesh from a mesh with manifold. While refining the boundary mesh the new points will be positioned according to a flat manifold with the result that the two meshes are inconsistent!?

**Comment by bangerth** [comment]
Shouldn't we just `clone()` the manifold?

**Comment by drwells** [comment]
I don't think that will work since we need a `Manifold<dim - 1, spacedim>` and that can't work for some manifolds, like `TransfiniteInterpolationManifold`.

**Comment by peterrum** [comment]
See also my comment https://github.com/dealii/dealii/pull/11566#issuecomment-761808020:

> The function GridGenerator::extract_boundary_mesh() does not attach any manifolds (as documented in the documentation). However, since the function might refine the mesh, manifolds have to be attached before calling the function even if they are only flat!

**Comment by bangerth** [comment]
Yes, @luca-heltai 's comment is how I think this needs to be handled.

**Comment by peterrum** [comment]
We already attach the flat manifolds:

https://github.com/dealii/dealii/blob/732766ac2383f48e6fc987aea25249e70514affd/source/grid/grid_generator.cc#L8540-L8549

Everything else seems to be wrong and very error-prone as long as we don't have submanifolds.
